src/framework/dispatch/common.rs

```diff
@@ -70,6 +70,27 @@ pub async fn check_permissions_and_cooldown<'a, U, E>(
         return Err(crate::FrameworkError::NotAnOwner { ctx });
     }
 
+    if cmd.guild_only && ctx.guild_id().is_none() {
+        return Err(crate::FrameworkError::GuildOnly { ctx });
+    }
+
+    if cmd.dm_only && ctx.guild_id().is_some() {
+        return Err(crate::FrameworkError::DmOnly { ctx });
+    }
+
+    if cmd.nsfw_only {
+        match ctx
+            .channel_id()
+            .to_channel(ctx.discord())
+            .await
+            .unwrap()
```

Comment:
Can you replace the unwrap? In case of an error, you can log a warning with `log::warn!` and return NsfwOnly error just to be safe